### PR TITLE
refactor: replace all calls to os.Exit() / CheckErr().

### LINKED ISSUE #1742: refactor all calls to os.Exit() / CheckErr()
in a number of places kubeadm calls:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/error.go#L54

if `err` is non-nil the function causes an abrupt `os.Exit(..)`.

we need to refactor all calls to this function and any stray calls to os.Exit().

mini-guide:
- clone kubernetes/kubernetes
- replace all Cobra `Run:` calls in kubeadm with `RunE` - e.g.:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/init.go#L139
- find all calls to `CheckErr` and replace them with the following pattern:
```
if err := someOperation(); err != nil {
   return err
}
```
- call CheckErr only here:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/kubeadm.go#L27
- run ./hack/verify-gofmt.sh, ./hack/update-bazel.sh
- also run `go test ./cmd/kubeadm/app/...` to make sure unit tests pass.
- commit changes
- send a PR

for more detail see the comments in this PR;
https://github.com/kubernetes/kubernetes/pull/80937#discussion_r312964179
https://github.com/kubernetes/kubernetes/pull/80937

cc @rosti for extra comments if needed.

### PR DESCRIPTION
**Pull request by jfbai**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Fix https://github.com/kubernetes/kubeadm/issues/1742
**Which issue(s) this PR fixes**:

Fixes #https://github.com/kubernetes/kubeadm/issues/1742

**Special notes for your reviewer**:

None

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

/assign @rosti 
/assign @neolit123

### CODE REVIEW

File: cmd/kubeadm/app/cmd/alpha/certs.go [hunk 1]
```diff
@@ -188,20 +192,24 @@ func addRenewFlags(cmd *cobra.Command, flags *renewFlags) {
 
 func renewCert(flags *renewFlags, kdir string, handler *renewal.CertificateRenewHandler) {
 	internalcfg, err := configutil.LoadOrDefaultInitConfiguration(flags.cfgPath, &kubeadmapiv1beta2.InitConfiguration{}, &flags.cfg)
-	kubeadmutil.CheckErr(err)
+	if err != nil {
+		return
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/alpha/certs.go]: we might have to return the actual error from this function here.

**Comment by neolit123** [comment]
thanks @jfbai 
this is looking good thus far, i've added one comment.

btw, the PR needs to wait until the ~10th of Sept. because we are in code freeze for 1.16 ATM.

File: cmd/kubeadm/app/cmd/alpha/certs.go (hunk 1 above)
Comment (jfbai) [reply cmd/kubeadm/app/cmd/alpha/certs.go]: Fixed.

**Comment by jfbai** [comment]
> thanks @jfbai
> this is looking good thus far, i've added one comment.
> 
> btw, the PR needs to wait until the ~10th of Sept. because we are in code freeze for 1.16 ATM.

@neolit123 Thanks a lot, I have fixed your comment.

**Review by rosti — COMMENTED** [review]
Thanks @jfbai !

I have just some style nits.
One additional thing, that we need to do is to add the following couple of lines, just before [this line](https://github.com/kubernetes/kubernetes/blob/c7c89f8c6140e0639dc8ba2d8dcdd6fdbd5409ae/cmd/kubeadm/app/cmd/cmd.go#L71):

```golang
SilenceErrors: true,
SilenceUsage: true,
```

This is because, the Cobra behavior changes slightly when `RunE` is used. If an error is returned it's dumped by Cobra and the command usage is displayed. Hence, to preserve the current behavior, we need to disable that.

**Comment by jfbai** [comment]
> Thanks @jfbai !
> 
> I have just some style nits.
> One additional thing, that we need to do is to add the following couple of lines, just before [this line](https://github.com/kubernetes/kubernetes/blob/c7c89f8c6140e0639dc8ba2d8dcdd6fdbd5409ae/cmd/kubeadm/app/cmd/cmd.go#L71):
> 
> ```go
> SilenceErrors: true,
> SilenceUsage: true,
> ```
> 
> This is because, the Cobra behavior changes slightly when `RunE` is used. If an error is returned it's dumped by Cobra and the command usage is displayed. Hence, to preserve the current behavior, we need to disable that.

@rosti Thanks for comments, I will fix them. :).

PS: previously, I returned the error in one line as you commented. Then, I thought that check `err` may help understanding and stay the `if` blocks.

**Comment by jfbai** [comment]
Hi @rosti , I fixed your comments, could you please take another look?

**Review by rosti — COMMENTED** [review]
Thanks @jfbai !
We just need to fix a few nits in the tests and this is ready to go (as soon as the master is unblocked again).

**Comment by jfbai** [comment]
@rosti PTAL, thanks a lot.

**Comment by jfbai** [comment]
@neolit123 @rosti Could this PR be merged now?

**Review by rosti — APPROVED** [review]
Looks great @jfbai !
Sorry for the delay, I was on a PTO.
/lgtm
/hold cancel

**Pull request merged**